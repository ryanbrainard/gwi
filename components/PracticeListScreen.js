import React from "react";
import PracticeDetailButton from "./PracticeDetailButton";
import { View } from "react-native";
import CharacterSet from "../models/CharacterSet";

export default class PracticeListScreen extends React.Component {
  static navigationOptions = {
    title: "Practice"
  };

  render() {
    const { navigation } = this.props;

    return (
      <View>
        {CharacterSet.all().map(charSet => (
          <PracticeDetailButton
            key={charSet.key}
            charSet={charSet}
            navigation={navigation}
          />
        ))}
      </View>
    );
  }
}
