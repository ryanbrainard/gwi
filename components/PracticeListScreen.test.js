import React from "react";
import PracticeListScreen from "./PracticeListScreen";
import renderer from "react-test-renderer";

test("renders correctly", () => {
  const tree = renderer.create(<PracticeListScreen />).toJSON();
  expect(tree).toMatchSnapshot();
});
