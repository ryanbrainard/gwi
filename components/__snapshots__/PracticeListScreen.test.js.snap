// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
      "justifyContent": "space-between",
      "margin": 100,
    }
  }
>
  <View
    accessibilityComponentType={undefined}
    accessibilityLabel={undefined}
    accessibilityTraits={undefined}
    accessible={true}
    collapsable={undefined}
    hasTVPreferredFocus={undefined}
    hitSlop={undefined}
    isTVSelectable={true}
    nativeID={undefined}
    onLayout={undefined}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#6495ED",
        "borderRadius": 75,
        "height": 75,
        "justifyContent": "center",
        "margin": 10,
        "opacity": 1,
        "width": 75,
      }
    }
    testID={undefined}
    tvParallaxProperties={undefined}
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "color": "black",
          "fontSize": 18,
          "fontWeight": "bold",
        }
      }
    >
      ㄱㄲㅋ
    </Text>
  </View>
  <View
    accessibilityComponentType={undefined}
    accessibilityLabel={undefined}
    accessibilityTraits={undefined}
    accessible={true}
    collapsable={undefined}
    hasTVPreferredFocus={undefined}
    hitSlop={undefined}
    isTVSelectable={true}
    nativeID={undefined}
    onLayout={undefined}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#6495ED",
        "borderRadius": 75,
        "height": 75,
        "justifyContent": "center",
        "margin": 10,
        "opacity": 1,
        "width": 75,
      }
    }
    testID={undefined}
    tvParallaxProperties={undefined}
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "color": "black",
          "fontSize": 18,
          "fontWeight": "bold",
        }
      }
    >
      ㄷㄸㅌ
    </Text>
  </View>
  <View
    accessibilityComponentType={undefined}
    accessibilityLabel={undefined}
    accessibilityTraits={undefined}
    accessible={true}
    collapsable={undefined}
    hasTVPreferredFocus={undefined}
    hitSlop={undefined}
    isTVSelectable={true}
    nativeID={undefined}
    onLayout={undefined}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#6495ED",
        "borderRadius": 75,
        "height": 75,
        "justifyContent": "center",
        "margin": 10,
        "opacity": 1,
        "width": 75,
      }
    }
    testID={undefined}
    tvParallaxProperties={undefined}
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "color": "black",
          "fontSize": 18,
          "fontWeight": "bold",
        }
      }
    >
      ㅂㅃㅍ
    </Text>
  </View>
  <View
    accessibilityComponentType={undefined}
    accessibilityLabel={undefined}
    accessibilityTraits={undefined}
    accessible={true}
    collapsable={undefined}
    hasTVPreferredFocus={undefined}
    hitSlop={undefined}
    isTVSelectable={true}
    nativeID={undefined}
    onLayout={undefined}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#6495ED",
        "borderRadius": 75,
        "height": 75,
        "justifyContent": "center",
        "margin": 10,
        "opacity": 1,
        "width": 75,
      }
    }
    testID={undefined}
    tvParallaxProperties={undefined}
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "color": "black",
          "fontSize": 18,
          "fontWeight": "bold",
        }
      }
    >
      ㅅㅆ
    </Text>
  </View>
  <View
    accessibilityComponentType={undefined}
    accessibilityLabel={undefined}
    accessibilityTraits={undefined}
    accessible={true}
    collapsable={undefined}
    hasTVPreferredFocus={undefined}
    hitSlop={undefined}
    isTVSelectable={true}
    nativeID={undefined}
    onLayout={undefined}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#6495ED",
        "borderRadius": 75,
        "height": 75,
        "justifyContent": "center",
        "margin": 10,
        "opacity": 1,
        "width": 75,
      }
    }
    testID={undefined}
    tvParallaxProperties={undefined}
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "color": "black",
          "fontSize": 18,
          "fontWeight": "bold",
        }
      }
    >
      ㅈㅉㅊ
    </Text>
  </View>
</View>
`;
